src/Init/Data/List/Basic.lean

```diff
@@ -902,4 +902,35 @@ theorem drop_eq_nil_of_le {as : List α} {i : Nat} (h : as.length ≤ i) : as.dr
   | _::_,  0   => simp at h
   | _::as, i+1 => simp at h; exact @drop_eq_nil_of_le as i (Nat.le_of_succ_le_succ h)
 
+/-- Auxiliary definition for `toArray`. -/
+@[inline_if_reduce]
+def toArrayAux : List α → Array α → Array α
+  | nil,       r => r
+  | cons a as, r => toArrayAux as (r.push a)
+
+/-- A non-tail-recursive version of `List.length`, used for `List.toArray`. -/
```

Comment:
Do we still need this? `List.length` is already non-tail-recursive. I think this dates back to when `List.lengthTR` was called `List.length`.

Reply:
The main current difference is that this has `inline_if_reduce` while `length` does not.

Just going by the name, that could likely be added to `length`, but I'm trying to be minimally disruptive until the new compiler development is resumed.